client/sites-dashboard/components/link-in-bio-banner/link-in-bio-banner.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import { useSelector } from 'react-redux';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { getPreference } from 'calypso/state/preferences/selectors';
+import { LINK_IN_BIO_BANNER_PREFERENCE } from './link-in-bio-banner-parts';
+import { LinkInBioBanners, BannerType } from './link-in-bio-banners';
+
+type Props = {
+	displayMode: 'row' | 'grid';
+	sites: SiteExcerptData[];
+};
+
+const hasLinkInBioSite = ( sites: SiteExcerptData[] ) => {
+	return Boolean(
+		sites.find( ( site ) => {
+			const option = site.options || {};
+			return option.site_intent === 'link-in-bio';
+		} )
+	);
+};
+
+export const LinkInBioBanner = ( props: Props ) => {
+	const { displayMode, sites } = props;
+	const siteCount = sites.length;
+	const doesNotAlreadyHaveALinkInBioSite = ! hasLinkInBioSite( sites );
+	const isMobile = useMobileBreakpoint();
+	const isBannerVisible = useSelector( ( state ) =>
+		getPreference( state, LINK_IN_BIO_BANNER_PREFERENCE )
+	);
+	const showBanner =
+		( doesNotAlreadyHaveALinkInBioSite && isBannerVisible == null ) || isBannerVisible;
+
+	const renderBanner = () => {
+		let bannerType: BannerType = 'none';
+		if ( showBanner ) {
+			if ( isMobile ) {
+				bannerType = 'tile';
+			} else if ( displayMode === 'row' ) {
+				if ( siteCount === 1 ) {
+					bannerType = 'row';
+				}
+			} else if ( displayMode === 'grid' ) {
+				if ( siteCount === 1 ) {
+					bannerType = 'double-tile';
+				} else if ( siteCount === 2 ) {
+					bannerType = 'tile';
+				}
+			}
+		}
+
+		return LinkInBioBanners[ bannerType ];
+	};
+
+	return renderBanner();
```

Comment:
I suggest removing the renderBanner function and returning the element directly. I also simplified the ifs a bit.

```suggestion
	let bannerType: BannerType = 'none';
	if ( showBanner ) {
		if ( isMobile ) {
			bannerType = 'tile';
		} else if ( displayMode === 'row' && siteCount === 1 ) {
			bannerType = 'row';
		} else if ( displayMode === 'grid' ) {
			if ( siteCount === 1 ) {
				bannerType = 'double-tile';
			} else if ( siteCount === 2 ) {
				bannerType = 'tile';
			}
		}
	}

	return LinkInBioBanners[ bannerType ];
```

Reply:
Yeah, the function wasn't adding anything here. Suggestion accepted.